tfdml/kernels/dml_diag_op.cc

```diff
@@ -95,17 +89,13 @@ class DmlDiagKernel : public DmlKernel
             0,
             0,
             0,
-            (num_elements + 1) * stride_multiplier,
+            (num_elements + 1),
         };
 
         DmlTensorInfo output;
         output.kernel_index = 0;
-        output.desc = DmlTensorDesc(
-            dtype_dml,
-            output_sizes,
-            output_strides,
-            0,
-            end_padding_in_bytes);
+        output.desc =
+            DmlTensorDesc(dtype_dml, output_sizes, output_strides, 0, 0);
```

Comment:
Nit: you can remove the values for `guaranteed_base_offset_alignment` and `end_padding_in_bytes` since their default value is already 0.